src/contracts/evm/ApplicationsMultiSig/index.ts

```diff
@@ -0,0 +1,53 @@
+import { Interface } from "@ethersproject/abi";
+import { BigNumber } from "@ethersproject/bignumber";
+import { TxLog } from "types/evm";
+import abi from "./abi.json";
+
+const iface = new Interface(abi);
+
+const confirmTransactionFn = iface.getFunction("confirmTransaction");
+const executeTransactionFn = iface.getFunction("executeTransaction");
+const revokeTransactionFn = iface.getFunction("revokeTransaction");
+
+export const executionTopic = iface.encodeFilterTopics("Execution", []);
+export const executionFailureTopic = iface.encodeFilterTopics(
+  "ExecutionFailure",
+  []
+);
+
+export const confirmTransaction = {
+  encode(transactionId: number) {
+    return iface.encodeFunctionData(confirmTransactionFn, [transactionId]);
+  },
+};
+
+export const executeTransaction = {
+  encode(transactionId: number) {
+    return iface.encodeFunctionData(executeTransactionFn, [transactionId]);
+  },
+  processLogs(logs: TxLog[]) {
+    const execTopic = iface.getEventTopic("Execution");
+    const execLog = logs.find((log) => log.topics.includes(execTopic));
+    if (!!execLog) {
+      const [id] = iface.decodeEventLog(
+        "Execution",
+        execLog.data,
+        execLog.topics
+      );
+      return (id as BigNumber).toString();
+    }
+    const execFailTopic = iface.getEventTopic("ExecutionFailure");
+    const execFailLog = logs.find((log) => log.topics.includes(execFailTopic));
+    if (!!execFailLog) {
+      throw new Error("ExecutionFailure");
```

Comment:
> Currently, `log / processLog` just parses logs and return something or nothin.

No reason for this function to remain unchanged if we find good reason for it. 
It would be more useful to throw different errors when they do appear than to return just "something or nothing". 

Why do you think it's better to have just these two results?

This also ties into the discussion about returning `TxSuccess.attrValue` VS returning transaction logs (see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2023#discussion_r1155542344). If we were to return logs in `TxSuccess`, it would make it possible to handle these different events/errors that get emitted.

Throwing errors which are handled by the calling component would allow for richer error messages both in our (console) logs and for the user.

Reply:
Let's say the called contract can emit `NotEnoughConfirmations` event (or error). We could do something like:
```
if `Execution` -> showMessageToUser("Transaction successful!");
else if `NotEnoughConfirmations` -> showMessageToUser("Not enough confirmations!");
else -> showMessageToUser("Transaction failed!");
```

Without these errors, there would be no way to know what exactly happened. We could still log them for dev purposes, sure, but it would be bad UX for the user to always show them "Transaction failed" when a more precise message could have been displayed, don't you agree @ap-justin @SovereignAndrey ?

Reply:
@misicnenad,
> It would be more useful to throw different errors when they do appear than to return just "something or nothing".

what were the different errors? To give an example in `endowmentCreation` context

1. I just sent and endowmentCreation msg and expecting `id` back (evm | polygon) doesn't matter - I already included in tx additional info that will retrieve that `id` (`string | undefined/null`)
2. I got back `attrValue`
3. just check if it's defined, `id ? "Success" : "Tx submitted but we didn't get id"

> Let's say the called contract can emit NotEnoughConfirmations event (or error). We could do something like:
if `Execution` -> showMessageToUser("Transaction successful!");
else if `NotEnoughConfirmations` -> showMessageToUser("Not enough confirmations!");
else -> showMessageToUser("Transaction failed!");

`Let's say the called contract can emit NotEnoughConfirmations event` - let's say this is actually the case and not just hypotheical scenario. If you are expecting this in your transaction you should just include it in the implementation of your log processor

1. get logs
2. in these logs, find an entry with event topic `NotEnoughConfirmations event`
3. if encountered, set your `attribute` value to say `"error"` else set to other value you expect from your logs `"something"`
4. then after tx check your `attrValue` against your expectation
```ts
attrValue === "error" ? "Hey something went wrong" : "Success" 
```

realistically though, ` can emit NotEnoughConfirmations event` low confirmation can be handled in `sendEVMTx/getReceipt` recursion

Reply:
In order not to hold the PR "hostage" to this discussion and treating this change as something to be discussed more, will just return `null` when the events weren't found.